.github/workflows/nightly_publish.yml

```diff
@@ -142,18 +142,31 @@ jobs:
           restore-keys: |
             ${{ runner.os }}-gradle-
       - name: Publish to snapshot repository
+        if: ${{ github.event.inputs.mode == "" || github.event.inputs.mode == "snapshot" }}
```

Comment:
There are only two possibilities for the input modes right? snapshot or staging. Why we need such complicated mechanism? By default snapshot is set.

Reply:
```
- name: Publish to snapshot/staging repository
  run: |
    ./gradlew publish -P${{ github.event.inputs.mode }}
    cd bom
    ./gradlew publish -P${{ github.event.inputs.mode }}
```

Reply:
@lanking520 If we have the inputs.mode as empty string, the publish will be empty. I think we'll need an if condition for an empty string and another one for the staging.

Reply:
We had a default value set on the top as:

```
default: 'snapshot'
```
Unless the person trigger the job with empty string purposefully this condition will be met. Another issue is if the user mispelled `staging` or `snapshot` what will happened? It does not throw any errors and just skip these publish stages which is not expected. If you really would like to handle the cases for a String with value other than staging/snapshot, you should do ${{ github.event.inputs.mode != "staging" && github.event.inputs.mode != "snapshot" }}